src/mqueryfront/src/Navigation.js

```diff
@@ -10,9 +10,13 @@ class Navigation extends Component {
     }
 
     render() {
+        const toHome = {
```

Comment:
Forced remount is not the correct way to fix that issue. Component can't reload correctly because there is no correct state transition from "job" mode to "query" mode.

QueryPage code has few issues related with that bug:
- state is derived from `this.props.match.params.hash` in incorrect way (this props should be used in all places where `qhash` is needed instead of putting it into initial state and relying only on state value. That's why we can't listen for path changes)
- "mode" should be probably deduced from "this.props.match.params.hash". It seems that view goes into the "job mode" only when route `query/:hash` is active.

QueryPage should be refactored to provide correct reaction for path change.

Reply:
This code just pass additional info to `QueryPage` (to  component and does not force remount.
With this additional information, the component can distinguish whether the path change has occurred from outside or from within.

Reply:
Follow-up from conversation: Ok, this is still useful when we want to clean-up Yara query form after clicking logo or 'Query' navitem